aiida/orm/autogroup.py

```diff
@@ -7,173 +7,286 @@
+    as accepted by CalculationFactory/DataFactory.
+    Each string contain the wildcard ``%`` at the end;
+    in this case this is used in a ``like`` comparison with the QueryBuilder.
+
+    Only one of the two (between exclude and include) can be set. If none of the two is set, everything is included.
     """
 
-    def _validate(self, param, is_exact=True):
-        """
-        Used internally to verify the sanity of exclude, include lists
-        """
-        from aiida.plugins import CalculationFactory, DataFactory
-
-        for i in param:
-            if not any([i.startswith('calculation'),
-                        i.startswith('code'),
-                        i.startswith('data'),
-                        i == 'all',
-            ]):
-                raise exceptions.ValidationError('Module not recognized, allow prefixes '
-                                      ' are: calculation, code or data')
-        the_param = [i + '.' for i in param]
-
-        factorydict = {'calculation': locals()['CalculationFactory'],
-                       'data': locals()['DataFactory']}
-
-        for i in the_param:
-            base, module = i.split('.', 1)
-            if base == 'code':
-                if module:
-                    raise exceptions.ValidationError('Cannot have subclasses for codes')
-            elif base == 'all':
-                continue
-            else:
-                if is_exact:
-                    try:
-                        factorydict[base](module.rstrip('.'))
-                    except exceptions.EntryPointError:
-                        raise exceptions.ValidationError('Cannot find the class to be excluded')
-        return the_param
+    def __init__(self):
+        """Initialize with defaults."""
+        self._exclude = None
+        self._include = None
+
+        now = timezone.now()
+        default_label_prefix = 'Verdi autogroup on ' + now.strftime('%Y-%m-%d %H:%M:%S')
+        self._group_label_prefix = default_label_prefix
+        self._group_label = None  # Actual group label, set by `get_or_create_group`
+
+    @staticmethod
+    def validate(strings):
+        """Validate the list of strings passed to set_include and set_exclude."""
+        if strings is None:
+            return
+        valid_prefixes = set(['aiida.node', 'aiida.calculations', 'aiida.workflows', 'aiida.data'])
+        for string in strings:
+            pieces = string.split(':')
+            if len(pieces) != 2:
+                raise exceptions.ValidationError(
+                    "'{}' is not a valid include/exclude filter, must contain two parts split by a colon".
+                    format(string)
+                )
+            if pieces[0] not in valid_prefixes:
+                raise exceptions.ValidationError(
+                    "'{}' has an invalid prefix, must be among: {}".format(string, sorted(valid_prefixes))
+                )
+
+            # If a % is present, it can only be the last character
```

Comment:
Is this just a rule that we decide, or an intrinsic one?
The following entry points are perfectly valid:
```
aiida.data:plugin.base.int
aiida.data:plugin.complex.int
```
so why wouldn't `aiida.data:plugin.%.int` not be a valid filter?

Reply:
I was thinking for simplicity to only consider the most common case. However I have now lifted this requirement.